connectors/spark/src/main/scala/io/unitycatalog/spark/UCSingleCatalog.scala

```diff
@@ -227,8 +245,17 @@ private class UCProxy extends TableCatalog with SupportsNamespaces {
     createTable.setName(ident.name())
     createTable.setSchemaName(ident.namespace().head)
     createTable.setCatalogName(this.name)
+
+    val hasExternalClause = properties.containsKey(TableCatalog.PROP_EXTERNAL)
     val storageLocation = properties.get(TableCatalog.PROP_LOCATION)
-    if (storageLocation == null) {
+    if (hasExternalClause && storageLocation == null) {
```

Comment:
This method is invoked after delta do commit 0 via the DeltaCatalog. Once DeltaCatalog wants to persist this table in UC, it calls this method which then uses SDK to create the table object in UC.

If the above is correct, shouldn't `storageLocation` be non-null for both managed / external?
- for external , it will be passed by user
- for managed, it will be auto generated by `UCSingleCatalog.createTable`

Reply:
This is for an invalid case `CREATE EXTERNAL TABLE ... [no LOCATION clause]`, see the test in https://github.com/unitycatalog/unitycatalog/pull/449/files#diff-3fc30b0d18b7b824cbdf24035313086c9b0787ab51392c62f910400b8081067eR329

Reply:
The overall flow is: `UCSingleCatalog#createTable` -> `DeltaCatalog#createTable` -> `CreateDeltaTableCommand#execute` -> `UCProxy#createTable` .

If user says `CREATE EXTERNAL TABLE ... [no LOCATION clause]`, then currently we will  end up creating a default path using [spark.sessionState.catalog.defaultTablePath(id)](https://github.com/delta-io/delta/pull/3663/files#diff-e880b2b407605f237c5bb566e3e06f981a4fbbb6ff675d7832b2cf5d2daa6b3dR146), we will initialize a delta log and then we will invoke `UCProxy#createTable` where we will fail.

Should we fail early i.e. in `UCSingleCatalog#createTable` itself when EXTERNAL is mentioned but location is not.

Reply:
+1 to failing early.